src/main/java/com/woowacourse/auth/presentation/LoginInterceptor.java

```diff
@@ -0,0 +1,43 @@
+package com.woowacourse.auth.presentation;
+
+import com.woowacourse.auth.application.JwtTokenProvider;
+import com.woowacourse.auth.exception.InvalidTokenException;
+import com.woowacourse.auth.exception.TokenNotFoundException;
+import com.woowacourse.auth.support.AuthorizationExtractor;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.springframework.http.HttpMethod;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerInterceptor;
+
+@Component
+public class LoginInterceptor implements HandlerInterceptor {
+
+    private final JwtTokenProvider jwtTokenProvider;
+    private final AuthenticationContext authenticationContext;
+
+    public LoginInterceptor(final JwtTokenProvider jwtTokenProvider,
+                            final AuthenticationContext authenticationContext) {
+        this.jwtTokenProvider = jwtTokenProvider;
+        this.authenticationContext = authenticationContext;
+    }
+
+    @Override
+    public boolean preHandle(final HttpServletRequest request, final HttpServletResponse response, final Object handler)
+            throws Exception {
+        if (isPreflight(request)) {
+            return true;
+        }
+        final String token = AuthorizationExtractor.extract(request)
+                .orElseThrow(TokenNotFoundException::new);
+        if (jwtTokenProvider.validateToken(token)) {
+            authenticationContext.setAuthority(jwtTokenProvider.getPayload(token));
+            return true;
+        }
+        throw new InvalidTokenException();
```

Comment:
이거 전에 말씀드린다고 했던 내용인데요 :)
`validationToken()`과 `getPayload()`는 둘다 jjwt의 resolving 메소드를 써요. 그럼 두 동작이 결국 boolean을 반환하느냐 payload를 반환하느냐의 차이일뿐 로직이 같다는 거죠.

그렇다면 동일한 두 로직이 똑같이 진행된다라는 생각이 드시지 않나요?

`getPayload()`에서 `JwtException`이 발생했을 때 `throw new InvalidTokenException();`를 해주면 아래와 같이 수정될 수도 있다라는 생각인데 어떻게 생각하세요?

```suggestion
        authenticationContext.setAuthority(jwtTokenProvider.getPayload(token));
        return true;
```

Reply:
동감합니다. `getClaimJws` 메서드까지 줄일 수 있을 것 같아요